src/storage/store/column.cpp

```diff
@@ -663,11 +662,10 @@ bool Column::canCommitInPlace(const ChunkState& state, const std::vector<offset_
     if (state.metadata.compMeta.canAlwaysUpdateInPlace()) {
         return true;
     }
-    for (auto i = 0u; i < dstOffsets.size(); i++) {
-        if (!state.metadata.compMeta.canUpdateInPlace(chunk->getData(), srcOffset + i,
-                dataType.getPhysicalType())) {
-            return false;
-        }
+    std::optional<NullMask> nullMask = chunk->getNullChunk()->getNullMask();
+    if (!state.metadata.compMeta.canUpdateInPlace(chunk->getData(), srcOffset, dstOffsets.size(),
```

Comment:
Should we just pass in the `std::optional<NullMask>` into `state.metadata.compMeta.canUpdateInPlace`?

Reply:
Oh, yes that would be simpler. I was thinking along the lines of the update code which takes a pointer since it may point to an existing nullmask, but here we don't need that.